File: src/main/java/thedarkcolour/futuremc/client/tesr/campfire/TESRCampfire.java
package thedarkcolour.futuremc.client.tesr.campfire;

import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.GlStateManager;
import net.minecraft.client.renderer.block.model.ItemCameraTransforms;
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.item.ItemStack;
import net.minecraft.util.EnumFacing;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import thedarkcolour.futuremc.block.BlockCampfire;
import thedarkcolour.futuremc.tile.TileCampfire;

@SideOnly(Side.CLIENT)
public class TESRCampfire extends TileEntitySpecialRenderer<TileCampfire> {
    @Override
    public void render(TileCampfire te, double x, double y, double z, float partialTicks, int destroyStage, float alpha) {
        EnumFacing facing = te.getWorld().getBlockState(te.getPos()).getValue(BlockCampfire.FACING);

        for(int i = 0; i < 4; ++i) {
            ItemStack stack = te.getBuffer().getStackInSlot(i);
            if (!stack.isEmpty()) {
                GlStateManager.pushMatrix();
                GlStateManager.translate(x + 0.5F, y + 0.44921875F, z + 0.5F);
                EnumFacing direction1 = EnumFacing.byHorizontalIndex((i + facing.getHorizontalIndex()) % 4);
                GlStateManager.rotate(-direction1.getHorizontalAngle(), 0.0F, 1.0F, 0.0F);
                GlStateManager.rotate(90.0F, 1.0F, 0.0F, 0.0F);
                GlStateManager.translate(-0.3125F, -0.3125F, 0.0F);
                GlStateManager.scale(0.375F, 0.375F, 0.375F);
                Minecraft.getMinecraft().getRenderItem().renderItem(stack, ItemCameraTransforms.TransformType.FIXED);
                GlStateManager.popMatrix();
            }
        }
    }
}

File: src/main/java/thedarkcolour/futuremc/client/tesr/campfire/TESRCampfire.kt
package thedarkcolour.futuremc.client.tesr.campfire

import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.GlStateManager
import net.minecraft.client.renderer.block.model.ItemCameraTransforms
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.util.EnumFacing
import thedarkcolour.futuremc.block.BlockCampfire.Companion.FACING
import thedarkcolour.futuremc.tile.TileCampfire

class TESRCampfire : TileEntitySpecialRenderer<TileCampfire>() {
    override fun render(te: TileCampfire, x: Double, y: Double, z: Double, partialTicks: Float, destroyStage: Int, alpha: Float) {
        val facing = try {
            te.world.getBlockState(te.pos).getValue(FACING)
        } catch (ignored: IllegalArgumentException) {
            EnumFacing.NORTH
        }
        for (i in 0..3) {
            val stack = te.inventory.getStackInSlot(i)
            if (!stack.isEmpty) {
                GlStateManager.pushMatrix()
                GlStateManager.translate(x + 0.5f, y + 0.44921875f, z + 0.5f)
                val direction1 = EnumFacing.byHorizontalIndex((i + facing.horizontalIndex) % 4)
                GlStateManager.rotate(-direction1.horizontalAngle, 0.0f, 1.0f, 0.0f)
                GlStateManager.rotate(90.0f, 1.0f, 0.0f, 0.0f)
                GlStateManager.translate(-0.3125f, -0.3125f, 0.0f)
                GlStateManager.scale(0.375f, 0.375f, 0.375f)
                Minecraft.getMinecraft().renderItem.renderItem(stack, ItemCameraTransforms.TransformType.FIXED)
                GlStateManager.popMatrix()
            }
        }
    }
}